src/test/java/io/confluent/idesidecar/restapi/application/SidecarInfoTest.java

```diff
@@ -54,134 +59,212 @@ public static SidecarInfo createSidecarInfo(TestInputs input) {
+                .formatted(
+                    sidecarInfo.version(),
+                    sidecarInfo.osArch())
         ),
         new TestInputs(
             "Windows 10 info without VS Code",
-            "Windows 10",
-            "10.0.1904562",
+            "windows",
+            "10.1.X.Y",
+            "20.1.2",
             OperatingSystemType.Windows,
             null,
-            null
+            null,
+            "Confluent-for-VSCode/vunknown (https://confluent.io; support@confluent.io) sidecar/v%s (windows/%s)"
+                .formatted(
+                    sidecarInfo.version(),
+                    sidecarInfo.osArch())
         ),
         // Windows 11
         new TestInputs(
             "Windows 11 info with VS Code",
-            "Windows 11",
-            "10.0.1904562",
+            "windows",
+            "10.1.X.Y",
+            "20.1.2",
             OperatingSystemType.Windows,
             new SidecarInfo.VsCode("20.1.2", "1.2.3"),
-            ""
+            "",
+            "Confluent-for-VSCode/v1.2.3 (https://confluent.io; support@confluent.io) sidecar/v%s (windows/%s)"
+                .formatted(
+                    sidecarInfo.version(),
+                    sidecarInfo.osArch())
         ),
         new TestInputs(
             "Windows 11 info with VS Code and version prefix",
-            "Windows 11",
-            "10.0.1904562",
+            "windows",
+            "10.1.X.Y",
+            "20.1.2",
             OperatingSystemType.Windows,
             new SidecarInfo.VsCode("20.1.2", "1.2.3"),
-            "v"
+            "v",
+            "Confluent-for-VSCode/v1.2.3 (https://confluent.io; support@confluent.io) sidecar/v%s (windows/%s)"
+                .formatted(
+                    sidecarInfo.version(),
+                    sidecarInfo.osArch())
         ),
         new TestInputs(
             "Windows 11 info without VS Code",
-            "Windows 11",
+            "windows",
             "10.1.X.Y",
+            "20.1.2",
             OperatingSystemType.Windows,
             null,
-            null
+            null,
+            "Confluent-for-VSCode/vunknown (https://confluent.io; support@confluent.io) sidecar/v%s (windows/%s)"
+                .formatted(
+                    sidecarInfo.version(),
+                    sidecarInfo.osArch())
         ),
         // Other
         new TestInputs(
-            "Solaris info with VS Code",
-            "Solaris 4",
+            "Unix info with VS Code",
+            "unix",
             "4.1.X.Y",
-            OperatingSystemType.Other,
+            "20.1.2",
+            OperatingSystemType.Unix,
             new SidecarInfo.VsCode("20.1.2", "1.2.3"),
-            ""
+            "",
+            "Confluent-for-VSCode/v1.2.3 (https://confluent.io; support@confluent.io) sidecar/v%s (unix/%s)"
+                .formatted(
+                    sidecarInfo.version(),
+                    sidecarInfo.osArch())
```

Comment:
Hmm, I think we shouldn't use the `sidecarInfo` instance to construct the expected output string (I alluded to this in my [earlier comment](https://github.com/confluentinc/ide-sidecar/pull/202#discussion_r1868306829)). By doing so, we'd be "baking in" the implementation into the tests. 

If we were to modify the implementation to return something other than a `String` from the `version()` or `osArch()` methods, we should expect these tests to break. However, since we're using the source implementation itself to construct the expected value, these tests would continue to pass. 

```suggestion
            "Confluent-for-VSCode/v1.2.3 (https://confluent.io; support@confluent.io) sidecar/v20.1.2 (unix/aarch64)"
```